### DISCUSSION TITLE
usb printer issue

### DISCUSSION THREAD

**Discussion by BroodTax** [ltsp/ltsp#739] [Q&A]
Hello! How to automatically connect a usb printer (kyocera p2040dn) when the client is turned on? (PPD driver installed in /usr/share/cups/model/Kyocera)

**Comment by alkisg** [comment]
Read this ltsp.conf option: http://ltsp.org/man/ltsp.conf/#CUPS_SERVER

> ...possible values are CUPS_SERVER="localhost", when a printer is connected to a client...

**Reply by BroodTax** [reply]
CUPS_SERVER="localhost" is set. I need to automatically connect the usb printer when the client is turned on, without having to set it to CUPS each time

**Reply by alkisg** [reply]
What does `set it to CUPS each time` mean?
What exactly do you need to do to make the printer work after the client boots?

**Reply by BroodTax** [reply]
Every time I restart the client, I have to set up the printer again, I would like to avoid this

**Reply by alkisg** [reply]
In Linux, USB printers are autodetected and autoinstalled.
That's why I'm asking **exactly** what actions you need to do each time to make it work, so that you do these things on the server instead, to affect the client image.

**Reply by BroodTax** [reply]
Every time I go to CUPS on client and set up a usb printer.
